### PR TITLE
feat(type annotation): added type info

### PR DESCRIPTION
**Pull request by louislewis2**
This will improve the quality of the generated `d.ts` file and also improves readability of the modified classes by specifying the expected type of parameters in the contructor and also for certain return values.

### CODE REVIEW

File: src/http-client.js [hunk 1]
```diff
@@ -3,13 +3,14 @@ import {Headers} from './headers';
 import {RequestBuilder} from './request-builder';
 import {HttpRequestMessage,createHttpRequestMessageProcessor} from './http-request-message';
 import {JSONPRequestMessage,createJSONPRequestMessageProcessor} from './jsonp-request-message';
+import { RequestMessageProcessor } from './request-message-processor';
```
Comment (bryanrsmith) [inline src/http-client.js]: Other imports in this project use no space around the curly braces, and it'd be nice to remain consistent.

File: src/headers.js [hunk 2]
```diff
@@ -3,19 +3,19 @@ export class Headers {
     this.headers = headers;
   }
 
-  add(key, value){
```
Comment (bryanrsmith) [inline src/headers.js]: It'd be great if type annotations were consistently filled in on all methods. Eg, this one is missing a type on `value`, and the return type.

File: src/http-client.js [hunk 3]
```diff
@@ -3,13 +3,14 @@ import {Headers} from './headers';
...
 
-function trackRequestStart(client, processor){
+function trackRequestStart(client : HttpClient, processor : RequestMessageProcessor){
```
Comment (bryanrsmith) [inline src/http-client.js]: These are private helper functions, so I don't think there's much value in annotating them.

File: src/headers.js (hunk 2 above)
Comment (louislewis2) [inline src/headers.js]: Bryan, would it be safe to assume that value is going to be of type string as well?

**Comment by louislewis2** [comment]
Hi @bryanrsmith thanks for the review & comments. I was intending on doing a bigger change in this repo, but thought I would start small to get a good handle on the process. For the private helper functions, I find value in having them typed as well, provides me and I am sure others with similar ide's a great design time experience.

**Comment by EisenbergEffect** [comment]
@louislewis2 Can you rebase on the master branch please? Thanks!

**Comment by louislewis2** [comment]
I have rebased from master. Please check if everything is in order now.

**Comment by EisenbergEffect** [comment]
Thats the way to do it! Merging shortly.

**Pull request merged**